### PR TITLE
Relay SDK Draft

### PR DESCRIPTION
**Pull request by v-Ajnava**
## Description

This is a Relay SDK first draft.
Swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/1061

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [X] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

File: src/ResourceManagement/Relay/Relay.Tests/project.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      "dependencies": {      
+      }
+    }
+  },
+  "dependencies": {
+    "Microsoft.NETCore.App": {
+      "type": "platform",
+      "version": "1.0.0"
+    },
+    "Microsoft.Rest.ClientRuntime.Azure.TestFramework": "1.5.0-preview",
```
Comment (shahabhijeet) [inline src/ResourceManagement/Relay/Relay.Tests/project.json]: @v-Ajnava can you use the latest Test Framework 1.6.0

File: src/ResourceManagement/Relay/Relay.Tests/Tests/ScenarioTests.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        protected void InitializeClients(MockContext context)
+        {
+            if (!m_initialized)
+            {
+                lock (m_lock)
+                {
+                    if (!m_initialized)
+                    {
+                        _resourceManagementClient = RelayManagementHelper.GetResourceManagementClient(context, new RecordedDelegatingHandler { StatusCodeToReturn = HttpStatusCode.OK });
+                        _relayManagementClient = RelayManagementHelper.GetServiceBusManagementClient(context, new RecordedDelegatingHandler { StatusCodeToReturn = HttpStatusCode.OK });
```
Comment (shahabhijeet) [inline src/ResourceManagement/Relay/Relay.Tests/Tests/ScenarioTests.cs]: @v-Ajnava please update the function name, make it GetRelayClient or something that says you are getting the relay client.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Comment by shahabhijeet** [comment]
Talked with @v-Ajnava and he mentioned more updated needs to be made to this PR due to swagger changes.

**Comment by shahabhijeet** [comment]
@v-Ajnava please remove "needs-revision" flag once this is ready to be reviewed.
Please add link to the swagger once the updated swagger is merged

File: src/ResourceManagement/Relay/Relay.Tests/project.json (hunk 1 above)
Comment (v-Ajnava) [inline src/ResourceManagement/Relay/Relay.Tests/project.json]: Updated.

File: src/ResourceManagement/Relay/Relay.Tests/Tests/ScenarioTests.cs (hunk 2 above)
Comment (v-Ajnava) [inline src/ResourceManagement/Relay/Relay.Tests/Tests/ScenarioTests.cs]: corrected the function name.

**Comment by v-Ajnava** [comment]
Updated Swagger Link : https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-relay/2016-07-01/swagger/relay.json

**Pull request merged**